lntest/itest/lnd_test.go

```diff
@@ -1353,23 +1369,20 @@ func testUnconfirmedChannelFunding(net *lntest.NetworkHarness, t *harnessTest) {
 	// Now, we'll connect her to Alice so that they can open a channel
 	// together. The funding flow should select Carol's unconfirmed output
 	// as she doesn't have any other funds since it's a new node.
+
 	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
 	if err := net.ConnectNodes(ctxt, carol, net.Alice); err != nil {
 		t.Fatalf("unable to connect dave to alice: %v", err)
 	}
-	ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
-	chanOpenUpdate, err := net.OpenChannel(
```

Comment:
Looking at `net.OpenChannel`, it seems that we already wait for the wallet to be synced for both nodes, why is the additional predicate in `openChannelStream` necessary?

Reply:
good question, i'm not sure. they call the same thing, but looks like it's still possible for it to flake w/o this

Reply:
Could be due to `synced_to_chain` flapping between true and false if the wallet hasn't caught up to the latest chain height.